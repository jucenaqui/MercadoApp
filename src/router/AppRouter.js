import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from 'react-router-dom';
import BaseApp from '../components/BaseApp';
import { ProductListApp } from '../components/product/ProductListApp';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch> 
                    <Route exact path="/" component={ BaseApp } />
                    <Route exact path="/items" component={ BaseApp } />
                    <Route exact path="/items/:id" component={ ProductListApp } />
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
