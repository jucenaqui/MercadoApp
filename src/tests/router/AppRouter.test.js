import { shallow } from "enzyme";
import { AppRouter } from "../../router/AppRouter";

describe('Test to <App />', () => {

    let wrapper = shallow(<AppRouter />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a Router component', () => {
        expect( wrapper.find('Router').exists() ).toBe(true);
    });

    test('should have a HeaderBarApp component', () => {
        expect( wrapper.find('HeaderBarApp').exists() ).toBe(true);
    });

});