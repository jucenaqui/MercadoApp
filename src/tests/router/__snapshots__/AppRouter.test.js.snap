// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <App /> should show the component succesfull 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <HeaderBarApp
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/items"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/items/:id"
      />
      <Redirect
        to="/"
      />
    </Switch>
  </div>
</Router>
`;
